russh/src/client/mod.rs

```diff
@@ -1111,11 +1110,11 @@ thread_local! {
 
 impl KexDhDone {
     async fn server_key_check<H: Handler>(
-        mut self,
+        &mut self,
```

Comment:
I'm not sure to understand why you made this change, since it implies a bunch of things to be cloned, while it seems the `KexDhDone` seems to be dropped after this call everytime anyways, I'd suggest still taking `self` and reverting the clone changes you did in `KexAlgorithm` and `KexDhDone::compute_keys`.

Reply:
You're right, it's not necessary, I reverted it